pkg/scaffold/v1/resource/resource.go

```diff
@@ -79,7 +80,10 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf(
 			"version must match ^v\\d+(alpha\\d+|beta\\d+)?$ (was %s)", r.Version)
 	}
-	if r.Kind != flect.Pascalize(r.Kind) {
+
+	validationErrors := validation.IsDNS1035Label(strings.ToLower(r.Kind))
+
+	if len(validationErrors) != 0 {
 		return fmt.Errorf("kind must be camelcase (expected %s was %s)", flect.Pascalize(r.Kind), r.Kind)
```

Comment:
there's still a call to `Pascalize` here; AFAICT this type of normalization should be eliminated.